src/scripting/lua_widget_attributes.cpp

```diff
@@ -231,9 +265,32 @@ WIDGET_SETTER("value", std::string, gui2::menu_button)
 	luaL_error(L, "attempt to modify read-only attribute 'value'");
 }
 
-WIDGET_SETTER("selected_index", int, gui2::menu_button)
+WIDGET_GETTER("item_count", int, gui2::multimenu_button)
 {
-	w.set_selected(value - 1, false);  // do not fire event, from lua this would only be used for setting default row
+	return w.get_item_count();
+}
+
+WIDGET_SETTER("item_count", int, gui2::multimenu_button)
+{
+	luaL_error(L, "attempt to modify read-only attribute 'item_count'");
+}
```

Comment:
I don't think this is needed? Just don't define a setter at all and you'll get an error like "unknown writable property: item_count".

Reply:
Needed?  No.  But I prefer to see an error message like this, as it tells me more than just "unknown modifiable value" (or something like that, pretty sure it doesn't actually even say "item_count", or if it does that just changed).  This message tells the user that a decision was made that this attribute is not writable (as compared to, say, they spelled the attribute name wrong or something).

BTW, this morning I made a spreadsheet of all the g/setters in widget_attrs.  I think there's a lot of things to fill in.

Reply:
So you can definitely delete this then, as I recently updated it to actually specify the unknown property in the error message. I'm not sure if you have that change in this branch, but it exists for sure.